### PR TITLE
Change samplers to use random state to allow consistency in reset par…

### PR DESCRIPTION
**Pull request by sankalp04**
Instead of sampling reset parameters directly from numpy, use the np.random_state to allow a fixed sequence of reset param values for a specified seed.

### CODE REVIEW

File: ml-agents/mlagents/trainers/learn.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def run_training(
     tc.start_learning(env, trainer_config)
 
 
-def create_sampler_manager(sampler_file_path, env_reset_params):
+def create_sampler_manager(sampler_file_path, env_reset_params, run_seed):
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/learn.py]: Maybe set the default for run_seed to None? Not that critical.

File: ml-agents-envs/mlagents/envs/sampler_class.py [hunk 2]
```diff
@@ -19,13 +19,19 @@ class UniformSampler(Sampler):
     """
 
     def __init__(
-        self, min_value: Union[int, float], max_value: Union[int, float], **kwargs
+        self,
+        min_value: Union[int, float],
+        max_value: Union[int, float],
+        seed: Optional[int] = None,
```
Comment (vincentpierre) [inline ml-agents-envs/mlagents/envs/sampler_class.py]: I do not think you can have optional fields followed by **kargs. If you do then every time there is **kargs you must provide a seed.

File: ml-agents-envs/mlagents/envs/sampler_class.py [hunk 3]
```diff
@@ -19,13 +19,19 @@ class UniformSampler(Sampler):
     """
 
     def __init__(
```
Comment (vincentpierre) [inline ml-agents-envs/mlagents/envs/sampler_class.py]: You need a docstring to explain what each argument does

File: ml-agents-envs/mlagents/envs/sampler_class.py (hunk 2 above)
Comment (ervteng) [reply ml-agents-envs/mlagents/envs/sampler_class.py]: I think default parameters followed by **kwargs are OK. But they have to be after *args. 

https://www.digitalocean.com/community/tutorials/how-to-use-args-and-kwargs-in-python-3

**Pull request merged**